### DISCUSSION TITLE
Feature Request: Add support for extension link to add custom autolink features.

### DISCUSSION THREAD

**Discussion by mann27** [ueberdosis/tiptap#5972] [Feature Requests]
### Description

The link should provide feature to add the custom logic to identify the links along with the default isAllowedUri. This provides adding go links and shortcuts like go/etc...

### Use Case

This would help support usecases where there are short cuts created with go links.

### Type

New feature

**Comment by nperez0111** [comment]
Is that different from the already existing `shouldAutoLink` callback option? https://tiptap.dev/docs/editor/extensions/marks/link#shouldautolink

**Reply by mann27** [reply]
It doesn't have the implementation of `defaultValidate` to be added and then to add the extra logic along with it.

**Reply by nperez0111** [reply]
You are going to have to be clearer here

**Reply by mann27** [reply]
Let me explain my usecase.

I need to identify the go/xyz in my editor and convert them to https://go/xyz.

So I want the shouldAutoLink to identify "go/" as a link ( also the default links should also be identified)  or I need to convert the go/xyz to an actual url so that by the default configuration of autolink it would be identified.

**Reply by nperez0111** [reply]
Nah, this is definitely scope creep for the link extension. What you are looking for instead is an input rule (specifically a markInputRule) https://tiptap.dev/docs/editor/extensions/custom-extensions/extend-existing#input-rules
